tasks/installation.yml

```diff
@@ -51,11 +51,36 @@
     - sap_preconfigure_min_package_check|bool
     - not( (__sap_preconfigure_min_pkgs is undefined) or (__sap_preconfigure_min_pkgs is none) or (__sap_preconfigure_min_pkgs | trim == '') )
 
-- name: ensure system is updated to the latest patchlevel
+- name: Ensure system is updated to the latest patchlevel
   package:
     state: latest
     name: "*"
   when: sap_preconfigure_update
   register: sap_preconfigure_register_packageinstall
 
+- set_fact:
+    needs_restarting_command: "needs-restarting -r"
+  when:
+    - ansible_os_family == 'RedHat'
+    - ansible_distribution_major_version == '7'
+
+- set_fact:
+    needs_restarting_command: "yum needs-restarting -r"
+  when:
+    - ansible_os_family == 'RedHat'
+    - ansible_distribution_major_version == '8'
+
+- name: Check if system needs to be restarted
+  shell: "{{ needs_restarting_command }}"
```

Comment:
use `command` instead of `shell` unless you need piping, output redirection, etc.

Reply:
Makes sense, thanks. I missed the description of the rc return value which is only linked from the sentence at the top of the table below Return Values in https://docs.ansible.com/ansible/latest/modules/command_module.html#command-module .

Reply:
solved in https://github.com/berndfinger/sap-preconfigure/commit/117fd8ff249a7f8fb29c78073329fb95a6365c58 .